Backend/TemporalDataManager.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime, timedelta
+import threading
+import time
+
+class TemporalDataManager:
+
+    # TODO: set up instance of algorithm manager (and anything else needed)
+    # TODO: tasked to set up threads that can access trends data retrieving functions
+    def __init__(self,  algomanager, loc, t=3):
+        self.algo = algomanager
+        self.timer = t
+        self.periodic_trends_on = True
+        self.locations = self.algo.get_all_locations()
+        self.timestamps = []
+        self.main_thread = threading.Thread(target=self.periodic_trends_retrieval, args=loc)
+        print("STARTING TEMPORAL THREAD")
+        self.testval = 0
+        self.main_thread.start()
+
+# continue here
+    # this method will be periodically called by a thread
+    # tasked to retrieve trends data and store them somewhere
+    # call functions to store snapshot entries in trends_snapshot databae table
+    #       table trends_snapshots(id, woe_id, trend_content, query_term, tweet_volume, is_hashtag, created_date)
+    def periodic_trends_retrieval(self, *loc):
+        while self.periodic_trends_on is True:
+            timestamp = datetime.now()
+            snapid = 0
+            for loc in self.locations:
+                restrends = self.algo.get_top_num_trends_from_location(int(loc['woeid']), 50)
+                print('TEMPORAL GET ', loc['woeid'])
+                snap = self.algo.create_trends_snapshot(snapid, restrends, loc, timestamp)
+
+                self.algo.add_trends_snapshot_to_database(snap)
+
+                # print(snap.__dict__)
+                snapid += 1
+                time.sleep(24)   # sleep for 24 seconds to avoid Twitter get/trends rate limit
+
```

Comment:
I didn't just mean that there's a timer between different locations' trends being fetched, but that there's a much larger interval between one time that you go check all of the locations' trends and the next time you check all of the locations' trends. This sleep call would go right AFTER the end of the for loop (i.e. not inside the for loop's scope) and would last for something like 1 or 3 hours.

Reply:
Otherwise we'll use up waaay too much database capacity and mostly be storing redundant data (because top trends often won't change that much within a 1-3 hour period)